src/cdk-experimental/listbox/listbox.ts

```diff
@@ -7,98 +7,214 @@
+    this._active = false;
   }
 
   static ngAcceptInputType_selected: BooleanInput;
+  static ngAcceptInputType_disabled: BooleanInput;
 }
 
-/**
- * Directive that applies interaction patterns to an element following the aria role of listbox.
- * Typically CdkOption elements are placed inside the listbox. Logic to handle keyboard navigation,
- * selection of options, active options, and disabled states is built in.
- */
 @Directive({
-  selector: '[cdkListbox]',
-  exportAs: 'cdkListbox',
-  host: {
-    role: 'listbox',
-  }
+    selector: '[cdkListbox]',
+    exportAs: 'cdkListbox',
+    host: {
+      'role': 'listbox',
+      '(keydown)': '_keydown($event)',
+      '[attr.aria-disabled]': '_disabled',
+    }
 })
-export class CdkListbox {
+export class CdkListbox implements AfterContentInit, OnDestroy {
+
+  _listKeyManager: ActiveDescendantKeyManager<CdkOption>;
+  private _disabled: boolean = false;
 
-  /** A query list containing all CdkOption elements within this listbox */
   @ContentChildren(CdkOption, {descendants: true}) _options: QueryList<CdkOption>;
 
   @Output() readonly selectionChange: EventEmitter<ListboxSelectionChangeEvent> =
       new EventEmitter<ListboxSelectionChangeEvent>();
 
-  /** Emits a selection change event, called when an option has its selected state changed */
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+
+  ngAfterContentInit() {
+    this._listKeyManager = new ActiveDescendantKeyManager(this._options)
+      .withWrap().withVerticalOrientation().withTypeAhead();
+  }
+
+  ngOnDestroy() {
+    this._listKeyManager.change.complete();
+  }
+
+  _keydown(event: KeyboardEvent) {
+    if (this._disabled) {
+      return;
+    }
+
+    const manager = this._listKeyManager;
+    const keyCode = event.keyCode;
+
+    if (keyCode === HOME || keyCode === END) {
+      event.preventDefault();
+      keyCode === HOME ? manager.setFirstItemActive() : manager.setLastItemActive();
+
+    } else if (keyCode === SPACE || keyCode === ENTER) {
+      if (manager.activeItem && !manager.isTyping()) {
+        this._toggleActiveOption();
+      }
+
+    } else {
+      manager.onKeydown(event);
+    }
+
+  }
+
+  /** Emits a selection change event, called when an option has its selected state changed. */
   _emitChangeEvent(option: CdkOption) {
     this.selectionChange.emit(new ListboxSelectionChangeEvent(this, option));
   }
 
-  /** Sets the given option's selected state to true */
+  private _toggleActiveOption() {
+    const activeOption = this._listKeyManager.activeItem;
+    if (activeOption && !activeOption.disabled) {
+      activeOption.toggle();
+      this._emitChangeEvent(activeOption);
+    }
+  }
+
+  /** Selects the given option if the option and listbox aren't disabled. */
   select(option: CdkOption) {
-    option.selected = true;
+    if (!this.disabled && !option.disabled) {
```

Comment:
It makes sense to prevent user selection when the control is disabled, but is it necessary to prevent programmatic selection as well?